client/client.go

```diff
@@ -220,3 +220,36 @@ func buildApiURL(server, url string) string {
 	}
 	return buildURL(server) + apiPrefix + url
 }
+
+// Used in tests.
+var exponentialBackoffSmallestUnit time.Duration = time.Minute
+
+// Simple algorithm: Start with one minute, and try three times, then double
+// interval (maxInterval is maximum) and try again. Repeat until we tried
+// three times with maxInterval.
+func GetExponentialBackoffTime(tried int, maxInterval time.Duration) (time.Duration, error) {
+	const perIntervalAttempts = 3
+
+	interval := 1 * exponentialBackoffSmallestUnit
+	nextInterval := interval
+
+	for c := 0; c <= tried; c += perIntervalAttempts {
+		interval = nextInterval
+		nextInterval *= 2
+		if interval >= maxInterval {
+			if tried-c >= perIntervalAttempts {
+				// At max interval and already tried three
+				// times. Give up.
+				return 0, errors.New("Tried maximum amount of times")
+			}
+
+			// Don't use less than one minute.
```

Comment:
Less than `exponentialBackoffSmallestUnit`. It is one minute now, but might be changed later, right.

Reply:
Right, I'll clarify this comment as well.